pkg/controller/provider/netlify/access.go

```diff
@@ -0,0 +1,170 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+*/
+
+package netlify
+
+import (
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/netlify/open-api/go/models"
+	"github.com/netlify/open-api/go/plumbing"
+	"github.com/netlify/open-api/go/plumbing/operations"
+	"k8s.io/client-go/util/flowcontrol"
+
+	"github.com/gardener/external-dns-management/pkg/dns/provider"
+	"github.com/gardener/external-dns-management/pkg/dns/provider/raw"
+)
+
+type Access interface {
+	ListZones(consume func(zone models.DNSZone) (bool, error)) error
+	ListRecords(zoneId string, consume func(record models.DNSRecord) (bool, error)) error
+
+	raw.Executor
+}
+
+type access struct {
+	client		operations.ClientService
+	authInfo	runtime.ClientAuthInfoWriter
+	metrics     provider.Metrics
+	rateLimiter flowcontrol.RateLimiter
+}
+
+func ClientCredentials(apiToken string) runtime.ClientAuthInfoWriter {
+	return runtime.ClientAuthInfoWriterFunc(func(r runtime.ClientRequest, _ strfmt.Registry) error {
+		r.SetHeaderParam("User-Agent", "external-dns-manager")
+		r.SetHeaderParam("Authorization", "Bearer "+apiToken)
+		return nil
+	})
+}
+
+func NewAccess(apiToken string, metrics provider.Metrics, rateLimiter flowcontrol.RateLimiter) (Access, error) {
+	client := plumbing.Default.Operations
+	clientCredentials := ClientCredentials(apiToken)
+	return &access{client: client, authInfo: clientCredentials, metrics: metrics, rateLimiter: rateLimiter}, nil
+}
+
+func (this *access) ListZones(consume func(zone models.DNSZone) (bool, error)) error {
+	this.metrics.AddRequests(provider.M_LISTZONES, 1)
+	this.rateLimiter.Accept()
+	results, err := this.client.GetDNSZones(nil, this.authInfo)
+	if err != nil {
+		return err
+	}
+	for _, z := range results.Payload {
+		if cont, err := consume(*z); !cont || err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (this *access) ListRecords(zoneId string, consume func(record models.DNSRecord) (bool, error)) error {
+	this.metrics.AddRequests(provider.M_LISTRECORDS, 1)
+	this.rateLimiter.Accept()
+	params := operations.GetDNSRecordsParams{ZoneID: zoneId}
+	results, err := this.client.GetDNSRecords(&params, this.authInfo)
+	if err != nil {
+		return err
+	}
+	for _, z := range results.Payload {
+		if cont, err := consume(*z); !cont || err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (this *access) CreateRecord(r raw.Record) error {
+	a := r.(*Record)
+	ttl := r.GetTTL()
+	testTTL(&ttl)
```

Comment:
May be this is a cloudflare specific behaviour (or a test artifact that hasn't been removed). I think/hope you don't need this.